core/icons/icon_types.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** The type for a mutator icon. Used for registration and access. */
+export const MUTATOR_TYPE = 'mutator';
+
+/** The type for a warning icon. Used for registration and access. */
+export const WARNING_TYPE = 'warning';
+
+/** The type for a warning icon. Used for registration and access. */
+export const COMMENT_TYPE = 'comment';
```

Comment:
Hmmm, I added this just to get around circular dependencies, (before I was just accessing `MutatorIcon.TYPE`), so idk that it's a good idea to commit to this further.

I think my ideal is that `getIcon`, `removeIcon`, etc would take in a `Type` class like we use for the registry, so that the   methods can use/return the proper subclass types.

I would like to attempt refactoring that before the end of the quarter, but I don't know that I'll have time! So for now I'll leave the types the same, mark these constants as internal, make sure they're not exported, and file an issue for the above.